codalab/worker/bundle_manager.py

```diff
@@ -381,6 +381,11 @@ def _construct_run_message(self, worker, bundle):
 
         # Parse |request_string| using |to_value|, but don't exceed |max_value|.
         def parse_and_min(to_value, request_string, default_value, max_value):
+            # On user-owned workers, ignore the maximum value. Users are free to
+            # use as many resources as they wish on their own machines.
+            if worker['user_id'] != self._model.root_user_id:
```

Comment:
We still need a limit - users don't want to get their machines accidentally hosed, so I think --max-memory should still take effect.